util/microservice-sdk/js/README.md

```diff
@@ -8,6 +8,132 @@ library scoped to attend the dojot microservices necessities.
+
+```bash
+ts -- sid -- LEVEL: [rid] message (file:line) {extra}
+```
+
+Where:
+
+* ts: timestamp in ISO format
+* sid: service identification
+* rid: request identification (optional metadata)
+* level: error, warn, info or debug
+* [rid]: request identifier (optional metadata)
+* message: message in string format
+* (file:line): file and line of the log message (optional metadata)
+* extra: any additional data - user specific (optional metadada)
+
+The messages are logged in the 'file' using the following format:
+
+```js
+{
+  level: string,
+  timestamp: <number>,
+  sid: <string>,
+  rid: <string>,
+  message: <string>,
+  file: <string>,
+  line: <number>,
+  extra: <object>,
+ }
+```
+
+Where:
+
+* ts: timestamp in unix format
+* sid: service identification
+* rid: request identification (optional metadata)
+* level: error, warn, info or debug
+* rid: request identifier (optional metadata)
+* message: message in string format
+* fine: file of the log message (optional metadata)
+* line: line of the log message (optional metadata)
+* extra: any additional data - user specific (optional metadada)
+
+The transports ('console' and 'file') are shared by all instances of the wrapper. So,
+any change in the transports will affect all instances.
+
+By default, no transport is set, so it is necessary to do this as part of the configuration
+or initialization of the microservice.
+
+It is usage is very simple and is shown by the following example:
+
+```js
+const { Logger } = require('@dojot/microservice-sdk');
+
+// By default, no transport is enabled; so you need to
+// set at least one before to log the messages.
+// It's worth to say that the transports are shared by all
+// modules of the microservice; consequently, any change
+// in the configuration will be valid for all!
+
+// Setting console transport
+// For more information about it, see:
+// https://github.com/winstonjs/winston/blob/HEAD/docs/transports.md#console-transport
+//
+// This is the expected transport to be used in a Docker container
+// once it is stateless.
+Logger.setTransport('console', {
+  // Any configuration put here will be merged with the defaults:
+  // { level: 'info',
+  //   format: consoleFormat /* customized format */}
+  level: 'debug',
+});
+
+// Setting file transport
+// For more information about it, see:
+// https://github.com/winstonjs/winston-daily-rotate-file#readme
+//
+// This transport should be used if you need to keep the logs in
+// files with rotation.
+// This is not typically the case for Docker container applications,
+// where the logs are expected to be redirected to /dev/stdout
+Logger.setTransport('file', {
+  // Any configuration put here will be merged with the defaults:
+  // { level: 'info',
+  //   dirname: '/var/log/',
+  //   filename: 'dojot.microservice-%DATE%.log',
+  //   datePattern: 'YYYY-MM-DD',
+  //   zippedArchive: true,
+  //   maxSize: '10m',
+  //   maxFiles: '7d',
```

Comment:
From the library documentation (https://github.com/winstonjs/winston-daily-rotate-file)

maxFiles: Maximum number of logs to keep. If not set, no logs will be removed. This can be a number of files or number of days. If using days, add 'd' as the suffix. (default: null)